docker/local/docker-compose.yml

```diff
@@ -0,0 +1,77 @@
+version: '3.5'
+
+services:
+  ## INITIALIZE GENESIS AND SECRETS
+  init:
+    build:
+      context: ../../
+      dockerfile: docker/local/Dockerfile
+    command: ["init"]
+    volumes:
+      - node-1:/data1
+      - node-2:/data2
+      - node-3:/data3
+      - node-4:/data4
+      - genesis:/genesis
+    network_mode: host
```

Comment:
Would it be possible to create a docker network for running this.  Removes the need to expose `libp2p` port on the host machine.  :thinking:

Reply:
Agreed. Containers should run in an internal network, and only `json-rpc` and maybe `grpc` should be exposed on the host.

Reply:
That will cause issues in Macos, most of us use Macos, and this in `genesis.json` won't work `/ip4/127.0.0.1/tcp/10001/p2p/`

Reply:
It would be much more difficult because the bootstrap node address must contain IP which is dynamic for docker networks. You cannot specify just an alias of the container.

Reply:
`/dns4/node-1/tcp/1478/p2p/` it won't pass an address validation because `node-1` is not an IP.

Reply:
The `dns4` multiaddr header instructs lib2p2 to resolve the hostname rather than using a static IP, which works out great for dynamic network interface scenarios like this. (Ref: https://wiki.polygon.technology/docs/edge/get-started/set-up-ibft-locally/#step-2-prepare-the-multiaddr-connection-string-for-the-bootnode) 

It's not a functional requirement for local, but overall a cleaner adoption.

Reply:
Good point, you are right, it works now. Updated docker-compose.

Reply:
Thanks. To finish up this change, we should be able to use the default respective ports `grpc`, `lib2p2`, and `json-rpc` within the container commands, and port map  `grpc` and `json-rpc` to use on the host. That way, users can send JSON-RPC commands to the nodes. :pray:

CC @vcastellm 

Example:
```yaml   
    command: ["server", "--data-dir", "/data", "--chain", "/genesis/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--seal"]
    ports:
      - '10000:9632'
      - '10002:8545'
```